presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/WindowFilterPushDown.java

```diff
@@ -266,23 +268,26 @@ private TopNRowNumberNode convertToTopNRowNumber(WindowNode windowNode, int limi
                     Optional.empty());
         }
 
-        private static boolean canReplaceWithRowNumber(WindowNode node)
+        private static boolean canReplaceWithRowNumber(WindowNode node, FunctionManager functionManager)
         {
-            return canOptimizeWindowFunction(node) && !node.getOrderingScheme().isPresent();
+            return canOptimizeWindowFunction(node, functionManager) && !node.getOrderingScheme().isPresent();
         }
 
-        private static boolean canOptimizeWindowFunction(WindowNode node)
+        private static boolean canOptimizeWindowFunction(WindowNode node, FunctionManager functionManager)
         {
             if (node.getWindowFunctions().size() != 1) {
                 return false;
             }
             Symbol rowNumberSymbol = getOnlyElement(node.getWindowFunctions().entrySet()).getKey();
-            return isRowNumberSignature(node.getWindowFunctions().get(rowNumberSymbol).getFunctionHandle().getSignature());
+            return isRowNumberMetadata(functionManager.getFunctionMetadata(node.getWindowFunctions().get(rowNumberSymbol).getFunctionHandle()));
         }
 
-        private static boolean isRowNumberSignature(Signature signature)
+        private static boolean isRowNumberMetadata(FunctionMetadata functionMetadata)
         {
-            return signature.equals(ROW_NUMBER_SIGNATURE);
+            return functionMetadata.getName().equals(ROW_NUMBER_SIGNATURE.getName())
```

Comment:
Can we have `ROW_NUMBER_FUNCTION_METADATA` and directly do `functionMetadata.equals(ROW_NUMBER_FUNCTION_METADATA)` ?

Reply:
The current situation is definitely not ideal. But clean up all signature related issues is an ongoing process (refer to be previous dozen commits). We'll eventually get them all.